### PR TITLE
Improve layered repo README.

### PR DESCRIPTION
**Pull request by hlinnaka**
Add an informal overview of how it works.

### CODE REVIEW

File: pageserver/src/layered_repository/README.md [hunk 1]
```diff
@@ -54,8 +75,52 @@ TODO: Are there any exceptions to this?
...
+
+A layer can be in different states:
+
+- Open - a layer where new WAL records can be appended to.
+- Closed - a layer that is read-only, no new WAL records can be appended to it
+- Historical: synonym for closed
+- InMemory: A layer that is kept only in memory, and needs to be rebuilt from WAL
+  on pageserver start
+- OnDisk: A layer that is stored on disk. If its end-LSN is older than
+  disk_consistent_lsn, it is known to be fully flushed and fsync'd to local disk.
```
Comment (lubennikovaav) [inline pageserver/src/layered_repository/README.md:89]: Do we ever use OnDisk layers that are not fully flushed?

File: pageserver/src/layered_repository/README.md [hunk 2]
```diff
@@ -54,8 +75,52 @@ TODO: Are there any exceptions to this?
...
+- Frozen layer: an in-memory layer that is Closed.
+
+There are two kinds of OnDisk layers:
+- ImageLayer represents an image or a snapshot of a segment at one particular LSN.
```
Comment (lubennikovaav) [inline pageserver/src/layered_repository/README.md:93]: The "segment" is a bit confusing here. Is it our "layer segment" or postgres relation 1Gb segment?

**Review by lubennikovaav — APPROVED** [review]

File: pageserver/src/layered_repository/README.md (hunk 1 above)
Comment (hlinnaka) [reply pageserver/src/layered_repository/README.md:89]: No, we wait for it to be fsync'd. https://github.com/zenithdb/zenith/pull/815/ will change that, though. After that, we will swap the layers in the layer map as soon as the on-disk layer has been written out to the file, but we don't immediately fsync it. That's done only later.

File: pageserver/src/layered_repository/README.md (hunk 2 above)
Comment (hlinnaka) [reply pageserver/src/layered_repository/README.md]: Changed it to "10 MB relish segment"

**Pull request merged**